### PR TITLE
Create tests for NerdTranslatorService

### PR DESCRIPTION
**Pull request by evgenysavonkin**
link: https://trello.com/c/1pLDVmZE

### CODE REVIEW

File: src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java
Comment (Electron3D) [inline src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java:1]: Would it be better to put repeating code in separate method with voice gender as parameter. New services don't needed on my mind.

File: src/test/java/com/nerdtranslator/translateapibridge/service/NerdTranslatorServiceTest.java
Comment (Electron3D) [inline src/test/java/com/nerdtranslator/translateapibridge/service/NerdTranslatorServiceTest.java:1]: Everything is good but maybe we can add naming design more official with providing more information in it.

**Review by Electron3D — CHANGES_REQUESTED** [review]

File: src/test/java/com/nerdtranslator/translateapibridge/service/NerdTranslatorServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+@SpringBootTest
+@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
+@RequiredArgsConstructor
+class NerdTranslatorServiceTest {
+
+    private final NerdTranslatorService nerdTranslatorService;
+
+    @Test
+    void oneEnglishWordShouldEqualToAnotherRussianWordCase() {
```
Comment (vamelchenia) [inline src/test/java/com/nerdtranslator/translateapibridge/service/NerdTranslatorServiceTest.java]: @Electron3D Do you think whether this and all other test cases names are in correct format? 
For me, they look clear enough but it's slightly different from the existing naming conventions, for example, here: https://dzone.com/articles/7-popular-unit-test-naming

Comment (Electron3D) [reply src/test/java/com/nerdtranslator/translateapibridge/service/NerdTranslatorServiceTest.java]: first of all it's integration tests, we've had a conversation with Eugeniy about test naming yesterday, my mentors on courses were teaching us to name integration tests kinda like that: "should_return_***_IT" or if we have unit tests something like "methodWeTest_ProvidedParams_ExpectedReturn"

File: src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java [hunk 2]
```diff
@@ -1,52 +1,105 @@
...
+            }
         }
         if (speechResult == null) {
-            throw new RuntimeException("The result of text to speech translation wasn't received");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
         }
+
         return speechResult;
     }
+
```
Comment (Electron3D) [inline src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java:52]: This method is enough for processing, replace try-catch in transformTextToSound with the code from it and add neutral gender into the list of voices.

File: src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java [hunk 3]
```diff
@@ -1,52 +1,105 @@
...
+    private byte[] isSuccessfulSpeechResultAgain(TextToSpeechClient textToSpeechClient, AudioEncoding audioEncoding,
+                                                 String textToTransfer, String langCode) {
+        byte[] result;
+        List<SsmlVoiceGender> genders = new ArrayList<>(List.of(SsmlVoiceGender.FEMALE, SsmlVoiceGender.MALE));
+        for (SsmlVoiceGender gender : genders) {
+            try {
+                result = getSpeechResult(textToSpeechClient, gender, audioEncoding, textToTransfer, langCode);
+                return result;
+            } catch (InvalidArgumentException e) {
+                continue;
```
Comment (Electron3D) [inline src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java:32]: throw new VoiceGenderNotSupportedException("\"Voice isn't supported yet\""); and add exception handling method in GlobalExceptionHandler with just logging, ot just log it here for now

File: src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java [hunk 4]
```diff
@@ -1,52 +1,105 @@
...
         }
         if (speechResult == null) {
-            throw new RuntimeException("The result of text to speech translation wasn't received");
+            throw new RuntimeException("\"The result of text to speech translation wasn't received\"");
         }
+
         return speechResult;
     }
+
+    private byte[] isSuccessfulSpeechResultAgain(TextToSpeechClient textToSpeechClient, AudioEncoding audioEncoding,
```
Comment (Electron3D) [inline src/main/java/com/nerdtranslator/translateapibridge/service/impl/TextToSpeechApiServiceImpl.java]: This method is enough for processing, replace try-catch in transformTextToSound with the code from it and add neutral gender into the list of voices.

**Review by Electron3D — COMMENTED** [review]
Comments in TextToSpeechApiServiceImpl

**Review by Electron3D — APPROVED** [review]

**Review by vamelchenia — APPROVED** [review]

**Pull request merged**